File: app/src/main/java/com/pawels96/skyrimperkcalculator/presentation/SkillAdapter.java
package com.pawels96.skyrimperkcalculator.presentation;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.pawels96.skyrimperkcalculator.R;
import com.pawels96.skyrimperkcalculator.domain.Skill;

import java.util.List;

import static com.pawels96.skyrimperkcalculator.presentation.Utils.getSkillName;

public class SkillAdapter extends BaseAdapter {

    //TODO replace with recycler adapter

    private List<Skill> skills;
    private LayoutInflater inflater;

    private Context c;

    public SkillAdapter(Context c, List<Skill> skills) {

        this.c = c;
        this.skills = skills;

        inflater = (LayoutInflater) c.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return skills.size();
    }

    @Override
    public Object getItem(int position) {
        return skills.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {

        final Holder holder;

        if (view == null) {
            view = inflater.inflate(R.layout.list_item_skill, parent, false);

            holder = new Holder();
            holder.name = view.findViewById(R.id.skill_name);
            holder.perks = view.findViewById(R.id.active_perks);

            view.setTag(holder);
        }
        else holder = (Holder) view.getTag();

        final Skill skill = skills.get(position);

        holder.name.setText(getSkillName(skill.getIskill(), c));
        int perks = skill.getSelectedPerksCount();
        String perksString = perks > 0 ? Integer.toString(perks) : "";
        holder.perks.setText(perksString);

        return view;
    }

    @Override
    public boolean isEnabled(int position) {
        return true;
    }

    private static class Holder {
        public TextView name;
        public TextView perks;
    }
}

File: app/src/main/java/com/pawels96/skyrimperkcalculator/presentation/dialogs/SkillAdapter.kt
package com.pawels96.skyrimperkcalculator.presentation.dialogs

import android.content.Context
import android.text.SpannableString
import android.text.SpannableStringBuilder
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.pawels96.skyrimperkcalculator.R
import com.pawels96.skyrimperkcalculator.domain.Skill
import com.pawels96.skyrimperkcalculator.domain.SkillType
import com.pawels96.skyrimperkcalculator.domain.enums.ESpecialSkill
import com.pawels96.skyrimperkcalculator.presentation.BuildAdapter
import com.pawels96.skyrimperkcalculator.presentation.Utils.getSkillName
import com.pawels96.skyrimperkcalculator.presentation.colored

class SkillAdapter(private val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>(){

    private val skills: MutableList<Skill> = mutableListOf()

    fun display(_skills : List<Skill>){
        this.skills.clear()
        this.skills.addAll(_skills)
        notifyDataSetChanged()
    }

    lateinit var onItemClick : (Int) -> Unit

    override fun getItemCount(): Int = skills.size

    override fun getItemViewType(position: Int): Int = R.layout.list_item_skill

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return SkillHolder(LayoutInflater.from(parent.context).inflate(viewType, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        Log.d("sizeee", position.toString())

        val hol = holder as SkillHolder
        val skill = skills[position]
        val perks = skill.selectedPerksCount
        val color = getColor(skill)
        hol.name.text = getSkillName(skill.iskill, context).colored(color)
        hol.perks.text = if (perks > 0) perks.toString().colored(color) else ""
        hol.root.setOnClickListener { onItemClick(position) }
    }

    private fun getColor(skill: Skill) : Int {

        val colorID =  when(skill.type){
            SkillType.COMBAT -> R.color.skillCombatBright
            SkillType.MAGIC -> R.color.skillMagicBright
            SkillType.STEALTH -> R.color.skillStealthBright
            SkillType.SPECIAL -> when(skill.iskill){
                ESpecialSkill.SKILL_LYCANTHROPY -> R.color.skillWerewolfBright
                ESpecialSkill.SKILL_VAMPIRISM -> R.color.skillVampireBright
                else -> R.color.colorFont
            }
        }

        return context.resources.getColor(colorID)
    }

    private class SkillHolder(val root: View) : RecyclerView.ViewHolder(root) {
        val name: TextView = root.findViewById(R.id.skill_name)
        val perks: TextView = root.findViewById(R.id.active_perks)
    }
}